Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -112,6 +106,10 @@ internal final class DatadogCore {
         self.contextProvider.subscribe(\.version, to: applicationVersionPublisher)
         self.contextProvider.subscribe(\.trackingConsent, to: consentPublisher)
 
+        // connect the core to the message bus.
+        // the bus will keep a weak ref to the core.
+        bus.connect(core: self)
```

Comment:
non-blocking/ It's is quite critical piece of logic we're changing here - modifying the way in which the heart 💜 of `DatadogCore` is connected to the neural systems of Features 🙃 (message bus). With the lack of any [`L`-size unit test](https://github.com/DataDog/dd-sdk-ios/wiki/Test-pyramid-in-Datadog-iOS-SDK-project) defined in [DatadogCoreTests](https://github.com/DataDog/dd-sdk-ios/blob/feature/v2/Tests/DatadogTests/Datadog/DatadogCore/DatadogCoreTests.swift) it looks like an open surgery. No test is guarding for zero regression in this case (we're changing message-bus related tests along with this new implementation, so they don't count).

This is non-blocking ask, but I'm raising a flag of having very limited unit tests coverage for `DatadogCore`. I'd be up for pairing on adding some that cover message-bus communication 🙂.

Reply:
I'm taking note on that, you are right 👍 We can brainstorm and pair on how to provide a good pattern for defining L-size units. I think the Integration Units target could be a good host for that. I will book a session!